public/app/features/alerting/unified/components/contact-points/useContactPoints.tsx

```diff
@@ -73,16 +74,36 @@ export function useContactPointsWithStatus() {
         ...result,
         contactPoints: result.data
           ? enhanceContactPointsWithMetadata(
-              result.data,
               fetchContactPointsStatus.data,
               fetchReceiverMetadata.data,
-              onCallMetadata
+              onCallMetadata,
+              result.data
             )
           : [],
       }),
+      skip: !getPoliciesCount,
     }
   );
 
+  // for Grafana Managed Alertmanager, we use the new read-only endpoint for getting the list of contact points
+  const fetchGrafanaContactPoints = alertmanagerApi.endpoints.getContactPointsList.useQuery(undefined, {
+    refetchOnFocus: true,
+    refetchOnReconnect: true,
+    selectFromResult: (result) => ({
+      ...result,
+      contactPoints: result.data
+        ? enhanceContactPointsWithMetadata(
+            fetchContactPointsStatus.data,
+            fetchReceiverMetadata.data,
+            onCallMetadata,
+            undefined, //no config data
+            result.data // contact points from the new readonly endpoint
+          )
+        : [],
+    }),
+    skip: getPoliciesCount || !isGrafanaAlertmanager,
+  });
+
   // we will fail silently for fetching OnCall plugin status and integrations
   const error = fetchAlertmanagerConfiguration.error ?? fetchContactPointsStatus.error;
   const isLoading =
```

Comment:
Shouldn't `isLoading` take `fetchGrafanaContactPoints` and `fetchAlertmanagerConfiguration` into account?